src/rpc/misc.cpp

```diff
@@ -958,6 +1003,157 @@ UniValue getspentinfo(const UniValue& params, bool fHelp)
     return obj;
 }
 
+UniValue gettotalsupply(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 0)
+        throw runtime_error(
+                "gettotalsupply\n"
+                        "\nReturns the total coin amount produced in the coinbase transactions up until the latest block.\n"
+                        "\nArguments: none\n"
+                        "\nResult:\n"
+                        "{\n"
+                        "  \"total\"  (string) The total supply in duffs\n"
+                        "}\n"
+                        "\nExamples:\n"
+                + HelpExampleCli("gettotalsupply", "")
+                + HelpExampleRpc("gettotalsupply", "")
+        );
+
+    CAmount total = 0;
+
+    if(!pblocktree->ReadTotalSupply(total))
+        throw JSONRPCError(RPC_DATABASE_ERROR, "Cannot read the total supply from the database");
+
```

Comment:
It will return bogus value if we upgraded from previous version and didn't do reindex. If `fAddressIndex` (or something similar) is required to get the total supply at least we can warn user here

Reply:
Done